x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log_handler.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TypeOf } from '@kbn/config-schema';
+import { RequestHandler } from 'kibana/server';
+import { AGENT_ACTIONS_INDEX } from '../../../../../fleet/common';
+import { EndpointActionLogRequestSchema } from '../../../../common/endpoint/schema/actions';
+
+import { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { EndpointAppContext } from '../../types';
+
+export const actionsLogRequestHandler = (
+  _endpointContext: EndpointAppContext
+): RequestHandler<
+  TypeOf<typeof EndpointActionLogRequestSchema.params>,
+  unknown,
+  unknown,
+  SecuritySolutionRequestHandlerContext
+> => {
+  return async (context, req, res) => {
+    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    let result;
+    try {
+      result = await esClient.search({
```

Comment:
This is a simple one, but if it get a bit more complicated we should consider moving the retrieval, status code validation of the result mapping to a service and leave the route handler as light as possible.

Reply:
I'm inclined to move it to a service already. Also cause I'm doing a bunch of refactoring already.